[NO-TICKET] Clarify Ruby profiler SSI support (#22110)
DataDog/documentation: content/en/profiler/enabling/ruby.md

Continuous Profiler is not supported on serverless platforms, such as AWS Lambda.

> [Single Step Instrumentation](https://docs.datadoghq.com/tracing/trace_collection/automatic_instrumentation/) for Kubernetes (using the Datadog Helm chart) is supported, but Single Step Instrumentation for Linux host or VM and Docker are not supported. You will still need to manually set the `DD_PROFILING_ENABLED=true` environment variable to enable profiling.

Author: > I'm not sure to understand: Linux host / VM+docker are not supported BUT it could work if DD_PROFILING_ENABLED=true?
> Or these are definitively not supported BUT DD_PROFILING_ENABLED=true is needed for Kubernetes?

Linux host/VM + docker never work AND kubernetes works if you additionally set `DD_PROFILING_ENABLED`. But clearly my text was confusing, so I've given another pass at it, let me know if you think it's clear now, or if it's still confusing.

Continuous Profiler is not supported on serverless platforms, such as AWS Lambda.

> [Single Step Instrumentation](https://docs.datadoghq.com/tracing/trace_collection/automatic_instrumentation/) for Linux host or VM and Docker are not supported.

Reviewer: ```suggestion
[Single Step Instrumentation](https://docs.datadoghq.com/tracing/trace_collection/automatic_instrumentation/) is not supported for Linux hosts, VMs, or Docker.
```

Author: Tweaked link, I think the doc structure slightly changed since I opened the PR:

```suggestion
[Single Step Instrumentation](https://docs.datadoghq.com/tracing/trace_collection/automatic_instrumentation/single-step-apm/) is not supported for Linux hosts, VMs, or Docker.
```

> Single Step Instrumentation for Kubernetes (using the Datadog Helm chart) is supported, but you will still need to manually set the `DD_PROFILING_ENABLED=true` environment variable to enable profiling.

Reviewer: ```suggestion
Single Step Instrumentation is supported for Kubernetes (using the Datadog Helm chart), but you need to manually set the `DD_PROFILING_ENABLED=true` environment variable to enable profiling.
```
